httpserver/service_node.cpp

```diff
@@ -268,9 +268,46 @@ void ServiceNode::send_sn_request(const std::shared_ptr<request_t>& req,
 
 void ServiceNode::register_listener(const std::string& pk,
                                     const std::shared_ptr<connection_t>& c) {
+
+    // NOTE: it is the responsibility of connection_t to deregister itself!
     pk_to_listeners[pk].push_back(c);
-    LOKI_LOG(debug, "register pubkey: {}, total pubkeys: {}", pk,
+    LOKI_LOG(debug, "Register pubkey: {}, total pubkeys: {}", pk,
              pk_to_listeners.size());
+
+    LOKI_LOG(debug, "Number of connections listening for {}: {}", pk,
+             pk_to_listeners[pk].size());
+}
+
+void ServiceNode::remove_listener(const std::string& pk,
+                                  const connection_t* const c) {
```

Comment:
Is there a reason for the raw pointer?

Reply:
I had a version of this patch where I call `remove_listener` from the destructor of `connection_t` (I might still do that in the future), and so I wouldn't be able to make a `shared_ptr` at that point. A raw pointer is not strictly necessary in this implementation, but I'm not sure if a smart pointer is any better in this case.

Reply:
I agree, no good reason to pass shared_ptr (which could suggest that the function wants to share ownership).
It would have simplified the `std::remove_if` to a `std::remove` but not worth changing.